source/System.cpp

```diff
@@ -303,6 +304,155 @@ void System::Load(const DataNode &node, Set<Planet> &planets)
+        }
+        else if(starName == "star/b5")
+        {
+            ramscoopModifiers.push_back(B5WIND);
+            solarPanelModifiers.push_back(B5BRIGHTNESS);
+        }
+        else if(starName == "star/f0")
+        {
+            ramscoopModifiers.push_back(F0WIND);
+            solarPanelModifiers.push_back(F0BRIGHTNESS);
+        }
+        else if(starName == "star/f5" || starName == "star/f5-old")
+        {
+            ramscoopModifiers.push_back(F5WIND);
+            solarPanelModifiers.push_back(F5BRIGHTNESS);
+        }
+        else if(starName == "star/g0" || starName == "star/g0-old")
+        {
+            ramscoopModifiers.push_back(G0WIND);
+            solarPanelModifiers.push_back(G0BRIGHTNESS);
+        }
+        else if(starName == "star/g5" || starName == "star/g5-old")
+        {
+            ramscoopModifiers.push_back(G5WIND);
+            solarPanelModifiers.push_back(G5BRIGHTNESS);
+        }
+        else if(starName == "star/giant")
+        {
+            ramscoopModifiers.push_back(GIANTWIND);
+            solarPanelModifiers.push_back(GIANTBRIGHTNESS);
+        }
+        else if(starName == "star/k0" || starName == "star/k0-old")
+        {
+            ramscoopModifiers.push_back(K0WIND);
+            solarPanelModifiers.push_back(K0BRIGHTNESS);
+        }
+        else if(starName == "star/k5" || starName == "star/k5-old")
+        {
+            ramscoopModifiers.push_back(K5WIND);
+            solarPanelModifiers.push_back(K5BRIGHTNESS);
+        }
+        else if(starName == "star/m0")
+        {
+            ramscoopModifiers.push_back(M0WIND);
+            solarPanelModifiers.push_back(M0BRIGHTNESS);
+        }else if(starName == "star/m4")
+        {
+            ramscoopModifiers.push_back(M4WIND);
+            solarPanelModifiers.push_back(M4BRIGHTNESS);
+        }else if(starName == "star/m8")
+        {
+            ramscoopModifiers.push_back(M8WIND);
+            solarPanelModifiers.push_back(M8BRIGHTNESS);
+        }
+        else if(starName == "star/nova")
+        {
+            ramscoopModifiers.push_back(NOVAWIND);
+            solarPanelModifiers.push_back(NOVABRIGHTNESS);
+        }
+        else if(starName == "star/wr")
+        {
+            ramscoopModifiers.push_back(WRWIND);
+            solarPanelModifiers.push_back(WRBRIGHTNESS);
+        }
+        else
+        {
+            ramscoopModifiers.push_back(1);
+            solarPanelModifiers.push_back(1);
+            // TOTO(Jeremy) get rid of this message statement.
+            Messages::Add("there is an unacounted for star type!");
+            Messages::Add(starName);
+        }
+        
+        starCount++;
+	}
+	ramscoopModifiers.sort();
+	solarPanelModifiers.sort();
+	
+	// Sum up modifiers (currently using 1/n to normalize the sum).
+	ramscoopEffectiveness = 0;
+	solarPanelEffectiveness = 0;
+	for(int i = 0; i < starCount; i++)
+    {
+        ramscoopEffectiveness += (ramscoopModifiers.back()/(i + 1));
```

Comment:
i don't think this is correct.

a better loop is:
```
for(int i = 0; i < starCount; ++i)
{
    rsE += rsM.back();
    spE += spM.back();
    rsM.pop_back();
    spM.pop_back();
}
rsE /= starCount;
spE /= starCount;
```
The loop as you have it takes 100% of the first value, then 50% of the 2nd value, and so on.

Furthermore:
I'm also not sure why you are even dividing here, since stellar outputs are directly additive (2x 100 lumens sources = 1x 200 lumen source). Perceived brightness is indeed not linear, but solar panels and matter collectors don't work on perception.

Not needing to do this loop (e.g. you correctly just add the values) means you do not even need the rsM and spM lists/maps/whatever, and in your if/elseif tree above you directly increment the solarWindStrength and luminosity variables.